File: golang/examples/count-words/count_words.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	var count = make(map[string]int)
	var text = "hello world how are you world and how are you"
	//fmt.Println(text)

	var words = strings.Split(text, " ")
	//fmt.Println(words)
	for _, word := range words {
		count[word]++
	}
	//fmt.Println(count)

	for k, v := range count {
		fmt.Printf("%s: %d\n", k, v)
	}

}

File: books/javascript/src/examples/objects/count_words.js
"use strict";

var text = 'this is a constructor this is not';

var words = text.split(/\s+/);
var count = {};
var i;
for (i in words) {
    var word = words[i];
    if (count[word] === undefined) {
        count[word] = 0;
    }
    count[word]++;
}
console.log(count);
// Object { this: 2, is: 2, a: 1, constructor: NaN, not: 1 }
